deepfence_server/reporters/completion/completion.go

```diff
@@ -43,14 +41,29 @@ func FieldValueCompletion[T reporters.Cypherable](ctx context.Context, req Compl
 	}
 	defer tx.Close()
 
-	query := `
-	MATCH (n:` + dummy.NodeType() + `)
-	WHERE n.` + req.FieldName + ` =~ '^` + req.Completion + `.*'
-	RETURN DISTINCT n.` + req.FieldName
+	query := ""
+
+	if req.ScanID != "" {
+		query = `
+		MATCH (n{node_id: $scan_id}) -[:DETECTED]-> (m) -[:IS]-> (r:` + dummy.NodeType() + `)
+		WHERE r.` + req.FieldName + ` =~ '^` + req.Completion + `.*'
+		WITH r.` + req.FieldName + ` as ` + req.FieldName + `, r.node_id as node_id ORDER BY node_id
```

Comment:
This `WITH` is not needed, you can move it to `RETURN`

Reply:
I tried with `RETURN`, but since it is with `DISTINCT`  I get error
```
In a WITH/RETURN with DISTINCT or an aggregation, it is not possible to access variables declared before the WITH/RETURN: r (line 3, column 51 (offset: 238))
"                RETURN DISTINCT r.cve_id ORDER BY r.node_id"
```

Reply:
What about doing the `ORDER BY` directly onto the `n.FieldName`?

Reply:
that should work out! Let me push the change